### ISSUE TITLE
Custom UmbracoApiControllers don't work after immediately Install / Upgrade

### ISSUE THREAD

**Issue by KevinJump** [umbraco/Umbraco-CMS#13309]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.3 

### Bug summary

If you have any custom UmbracoApiController code _(note not UmbracoAuthorizedApiController they work)._ 

Then immeditly after a fresh install or upgrade of Umbraco these controllers don't work instead you get redirected to `https://mysite/account/accessdenied`

restarting the site once after install/reboot and it all works .

### Specifics

I tried to see if i could find something in the code that might not be getting fired when Umbraco 'soft restarts' after an install or upgrade but i got lost in the code ☹️ 

_**If someone could give me a pointer at where in the code the checks might be happening I will happily take a deeper look**_

### Steps to reproduce

1. Add a sample UmbracoApiController to your site. 

```cs
using Microsoft.AspNetCore.Mvc;
using Umbraco.Cms.Web.Common.Controllers;

namespace SampleEndPoints;

public class OneController : UmbracoApiController
{
    /// <summary>
    ///  for example should work on https://localhost:port/Umbraco/Api/One/GetApi
    /// </summary>
    /// <returns></returns>
    [HttpGet]
    public string GetApi() => "One";
}
```

2. Run a fresh install or an upgrade of that site, 
3. Visit the URL - you get redirected 
![image](https://user-images.githubusercontent.com/431231/198269162-91e7a305-d151-4b9d-9c58-5b8109ba4242.png)

4. Restart the site 
5. Visit URL - you get a response from the controller 

![image](https://user-images.githubusercontent.com/431231/198269353-39a3a853-b6f2-490e-98a6-dbf24d665365.png)

### Expected result / actual result

UmbracoApiControllers  (not backoffice ones they work) - should work on after an upgrade or install without the need to reboot. 

---
_This item has been added to our backlog [AB#24726](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/24726)_

**Comment by Zeegaan** [comment]
I can reproduce this, this does seem like an odd one 😅

**Comment by nikolajlauridsen** [comment]
This has been fixed by #13551, thanks for reporting 😄🐛🕵️

**Comment by Aladdin4u** [comment]
I'm having this same issue, it's working on my local server after hosting it on smartAsp.net it stopped working
